cilium/CFP-28175-BGP-v2-API-changes.md

```diff
@@ -0,0 +1,895 @@
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+}
+
+```
+
+
+```yaml
+---
+apiVersion: "cilium.io/v2alpha1"
+kind: CiliumBGPClusterConfig
+metadata:
+  name: 01-cilium-bgp-config
+spec:
+  nodeSelector:
+    matchLabels:
+      rack: rack1
+  bgpInstances:
+  - localASN: 65001
+    peers:
+    - peerAddress: "10.1.1.1"
+      peerASN: 65002
+      peerConfigRef:
+        name: "rack1"
+    - peerAddress: "10.1.1.2"
+      peerASN: 65002
+      peerConfigRef:
+        name: "rack1"
+```
+
+## CiliumBGPAdvertisement
+
+CiliumBGPAdvertisement is a resource that defines which routes to advertise to BGP peers. Multiple advertisements can be defined in the CiliumBGPAdvertisementSpec.  Same advertisement can also be referenced by multiple peer configuration resources.
+
+Each advertisement will have the following fields:
+* SelectorType: This field defines the selector type. Currently supported types are
+    * PodCIDR
+    * CiliumPodIPPool
+    * CiliumLoadBalancerIPPool
+
+    This field is not closed to change, consumers of CiliumBGPAdvertisement should handle unknown fields.
+
+* Selector: This field defines the label selector for the resource type defined in the SelectorType field. **If nil or empty, all prefixes of selector type will be advertised.**
+* Attributes: This defines the BGP attributes that are associated with this advertisement. These are standard BGP-specific attributes.
+
+CiliumBGPAdvertisement can be referenced from peer address family configuration. BGP instances will advertise defined NLRI per peer per family.
+
+**Relationship with other resources**
+
+Referenced by
+* CiliumBGPPeerConfig: CiliumBGPAdvertisement is referenced from the address family configuration of CiliumBGPPeerConfig.
+
+Multiple peer configurations can refer to the same advertisement. This is useful when we want to advertise the same routes to multiple peers which may have different peer configurations.
+
+**Caveats and future improvements**
+
+The current design assumes that all prefixes of one type are advertised with the same set of attributes, even when they are advertised individually (as host routes). There may be a need to advertise subsets of these prefixes with different attributes, e.g. cluster with a single LBIPPool but with multiple tenants where each tenant needs to be identified with a unique set of attributes (e.g. BGP communities). Some of the potential solutions to this may be:
+
+* Split LBIPPool into smaller non-overlapping ranges. This assumes that users (tenants) know and use the right pool
+* Per-service selector in AdvertisementType, e.g.
+
+```yaml
+        advertisedPathAttributes:
+         - selectorType: Service
+           selector:
+             matchLabels:
+               environment: production
+```
+
+**Consumer**
+
+This resource will be consumed by the BGP controller running inside the Cilium agent.
+
+**Status**
+
+No status field planned for this resource.
+
+```go
```

Comment:
There has been update to BGP Advertisement CRD, see [this](https://github.com/cilium/cilium/pull/30902) for more details. I don't know if this CFP need to be upto date with current development as things are always changing slightly.